src/cargo/ops/fix.rs

```diff
@@ -582,7 +588,11 @@ impl Default for PrepareFor {
 impl FixArgs {
     fn get() -> FixArgs {
         let mut ret = FixArgs::default();
-        ret.rustc = env::args_os().nth(1).map(PathBuf::from);
+        if env::var(CLIPPY_FIX_ENV).is_ok() {
+            ret.rustc = Some(PathBuf::from("clippy-driver"));
```

Comment:
Will there be some error handling when clippy-driver is not available?

Reply:
now that you mention it, yes there will. :)

Reply:
I tried and failed to figure out where the "To install" line comes from, though it seems that without any changes, just trying to invoke clippy-driver gives us a useful error message.

![Screenshot from 2019-06-24 19-25-12](https://user-images.githubusercontent.com/1993852/60064666-f8dd2400-96b5-11e9-9873-921d94751416.png)

alternatively I can add the following

```diff
diff --git a/src/cargo/ops/fix.rs b/src/cargo/ops/fix.rs
index e939dba3..8d324e6e 100644
--- a/src/cargo/ops/fix.rs
+++ b/src/cargo/ops/fix.rs
@@ -105,6 +105,11 @@ pub fn fix(ws: &Workspace<'_>, opts: &mut FixOptions<'_>) -> CargoResult<()> {
     }

     if opts.use_clippy {
+        // if let Err(e) = util::process("clippy-driver").arg("-V").exec_with_output() {
+        //     return Err(failure::format_err!(
+        //             "clippy-driver not available, please install clippy component\n{:?}", e
+        //     ).into());
+        // }
         wrapper.env(CLIPPY_FIX_ENV, "1");
     }

```

Reply:
I think that if cargo had its own "no-clippy!"-error message, that might be the saver variant, in case something at rustup changes and we no longer get the warning.

Reply:
I added a warning message but I didn't make it an Err because I would like to still get the nicer formatted rustup error message if its available.